EIPS/eip-4906.md

```diff
@@ -0,0 +1,56 @@
+---
+eip: 4906
+title: ERC-721/ERC-1155 Metadata Update Extension
+description: Standard interface extension for ERC-721/ERC-115 metadata update
+author: Anders (@0xanders), Lance (@LanceSnow)
+discussions-to: https://ethereum-magicians.org/t/eip4906-erc-721-erc-1155-metadata-update-extension/8588
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-03-13
+requires: 721, 1155
+---
+
+## Abstract
+
+This specification defines standard metadata update event of [ERC-721](./eip-721.md) and [ERC-1155](./eip-1155.md). The proposal depends on and extends the existing [ERC-721](./eip-721.md) and [ERC-1155](./eip-1155.md).
+
+## Motivation
+
+Many [ERC-721](./eip-721.md) and [ERC-1155](./eip-1155.md) contracts emit their custom event when metadata changed. It is easy to update metadata of one NFT by specific event, but it is difficult for third-party platforms such as NFT marketplace to update metadata of many NFTs based on custom events.
+
+Having a standard `MetadataUpdate` event will make it easy for third-party platforms to timely update metadata of many NFTs.
+
+## Specification
+
+The keywords “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+The **metadata update extension** is OPTIONAL for ERC-721/ERC-1155 contracts.
+
+
+```solidity
+/// @title ERC-721/ERC-1155 Metadata Update Extension
+interface IERC4906  {
+    /// @dev This event emits when the metadata of a token is changed.
+    /// So that the third-party platforms such as NFT market could
+    /// timely update the images and related attributes of the NFT
+    event MetadataUpdate(uint256 indexed _tokenId);
+}
+```
+
+The `MetadataUpdate` event MUST be emitted when the metadata of a token is changed.
+
+## Rationale
+
+Different NFTs have different metadata, and metadata generally has multiple fields. `bytes data` could be used to represents the modified value of metadata.  It is difficult for third-party platforms to identify various types of `bytes data`, so there is only one parameter `uint256 indexed _tokenId` in `MetadataUpdate` event.
+
+After capturing the `MetadataUpdate` event, a third party can update the metadata with information returned from the `tokenURI(uint256 _tokenId)` of ERC721 or `uri(uint256 _tokenId)` of ERC1155.
+
+## Backwards Compatibility
+
+No issues.
+
```

Comment:
There needs to be a Security Considerations section, see [EIP-1](https://eips.ethereum.org/EIPS/eip-1)